src/openms/source/ANALYSIS/OPENSWATH/MRMFeatureSelector.cpp

```diff
@@ -32,8 +32,13 @@
 // $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
 // --------------------------------------------------------------------------
 
+#include <algorithm>
+#include <unordered_set>
+#include <OpenMS/KERNEL/FeatureMap.h>
+#include <OpenMS/DATASTRUCTURES/LPWrapper.h>
 #include <OpenMS/ANALYSIS/OPENSWATH/MRMFeatureSelector.h>
 
+
 namespace OpenMS
 {
   MRMFeatureSelector::MRMFeatureSelector() :
```

Comment:
I would recommend splitting this into two classes `MRMFeatureSelectorQMIP` for the QMIP algorithm (encapsulated in methods `select_MRMFeatures_qmip` and `optimize_Tr`) and another class `MRMFeatureSelectorLP` for the baseline LP algorithm (encapsulated in methods `select_MRMFeatures_score` and `optimize_scores`).

Reply:
If there end up being enough shared methods between `MRMFeatureSelectorQMIP` and `MRMFeatureSelectorLP` we could also make a base class (e.g., `MRMFeatureSelector`) that encapsulates the shared methods between them.